src/graph/PropertyRetriever.ts

```diff
@@ -24,7 +25,8 @@ export default class PropertyRetriever {
         + 'You should probably provide a unique name to one of the providers: @Provide({name: \'uniqueName\')})',
       );
     }
-    return undefined;
+    const errorMessage = new UnresolvedDependencyExceptionFactory().generateErrorMessage(property, this.graph);
```

Comment:
We should throw only if the property name doesn't start with an underscore. If the property starts with an underscore we should return `undefined`.